application/client/src/app/ui/views/sidebar/search/providers/providers.ts

```diff
@@ -500,6 +506,67 @@ export class Providers {
         this.subjects.get().context.emit(event);
     }
 
+
+    public contextMenuOptions(items: IMenuItem[]): IMenuItem[] {
+        const historySession = history.get(this.session);
+        if(historySession === undefined) {
+            this.logger.error('History session is not defined');
+            return items
+        }
+        items.push({ /* Delimiter */ });
+        const store = this.session.search.store();
+        const showExport: boolean = (store.filters().get().length + store.charts().get().length + store.disabled().get().length) !== 0;
+        showExport && items.push(
+        {
+            caption: 'Export All to File',
+            handler: () => this._exportFilters(historySession),
+        });
+        items.push(
+        {
+            caption: 'Import from File',
+            handler: () => this._importFilterFile(historySession),
+        });
+        return items;
+    }
+
+    private _exportFilters = (historySession: HistorySession) => {
```

Comment:
I would group methods `_exportFilters` and `_importFilterFile` into something like:
```
protected history(historySession: HistorySession): { import(): void; export(): void } => {
    return {
        export: (): void => {...},
        import: (): void => {...},
    };
}
```

Reply:
But then wouldn't it create confusion with https://github.com/esrlabs/chipmunk/blob/master/application/client/src/app/service/history.ts#L75?

Reply:
How aboout `protected filters....`